src/fastmcp/server/http.py

```diff
@@ -284,18 +284,19 @@ def create_streamable_http_app(
     server_middleware: list[Middleware] = []
 
     # Create session manager using the provided event store
-    session_manager = StreamableHTTPSessionManager(
-        app=server._mcp_server,
-        event_store=event_store,
-        json_response=json_response,
-        stateless=stateless_http,
-    )
+    if server._session_manager is None:
+        server._session_manager = StreamableHTTPSessionManager(
+            app=server._mcp_server,
+            event_store=event_store,
+            json_response=json_response,
+            stateless=stateless_http,
+        )
```

Comment:
I'm hesitant about setting the private `_session_manager` outside of the class,
but it seems impossible without refactoring since it's currently separated into a different function